src/components/appointmentForm/AppointmentForm.jsx

```diff
@@ -0,0 +1,157 @@
+import dayjs from 'dayjs';
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect, useState } from 'react';
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import Select from '@mui/material/Select';
+import { DemoContainer } from '@mui/x-date-pickers/internals/demo';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { TimePicker } from '@mui/x-date-pickers/TimePicker';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { fetchDoctors } from '../../redux/doctor/doctorSlice';
+import { createAppointment } from '../../redux/appointment/appointmentsSlice';
+import cities from '../Cities';
+
+function BookAppointment() {
+  const { name, id } = useLocation()?.state?.doctor || {};
+  const navigate = useNavigate();
+  const [selectedDoctor, setSelectedDoctor] = useState('');
+  const [doctorId, setDoctorId] = useState();
+  const [selectedCity, setSelectedCity] = useState('');
+  const [selectedDate, setSelectedDate] = useState(dayjs());
+  const [selectedTime, setSelectedTime] = useState(dayjs());
+
+  // Ensure fetchedDoctors is initialized as an array
+  const fetchedDoctors = useSelector((state) => state.doctors.doctors) || [];
+
```

Comment:
Thank you for implementing the appointment booking feature. However, I noticed that when attempting to book an appointment, the doctor's name isn't displayed. Could you please address this issue?

![now showing doctor](https://github.com/mahdinoori2000/healthcare-app-frontend/assets/126486418/0515675d-da2f-4264-8de1-dc3b6c0600a8)